cmd/node_agent/na/gcp_test.go

```diff
@@ -27,28 +28,93 @@ const (
 
 // mockTokenFetcher implements the mock token fetcher.
 type mockTokenFetcher struct {
+	token        string
+	errorMessage string
 }
 
 // A mock fetcher for FetchToken.
 func (fetcher *mockTokenFetcher) FetchToken() (string, error) {
-	return token, nil
+	var err error
```

Comment:
You don't need to claim var. Return error directly in the if statement:
if len(fetcher.errorMessage) > 0 {
		return nil, fmt.Errorf(fetcher.errorMessage)
}

return token, nil

Reply:
Done